beacon_node/network/src/network_beacon_processor/sync_methods.rs

```diff
@@ -167,26 +168,33 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
         metrics::inc_counter(&metrics::BEACON_PROCESSOR_RPC_BLOCK_IMPORTED_TOTAL);
 
         // RPC block imported, regardless of process type
-        if let &Ok(AvailabilityProcessingStatus::Imported(hash)) = &result {
-            info!(self.log, "New RPC block received"; "slot" => slot, "hash" => %hash);
+        match result.as_ref() {
+            Ok(AvailabilityProcessingStatus::Imported(hash)) => {
+                info!(self.log, "New RPC block received"; "slot" => slot, "hash" => %hash);
 
-            // Trigger processing for work referencing this block.
-            let reprocess_msg = ReprocessQueueMessage::BlockImported {
-                block_root: hash,
-                parent_root,
-            };
-            if reprocess_tx.try_send(reprocess_msg).is_err() {
-                error!(self.log, "Failed to inform block import"; "source" => "rpc", "block_root" => %hash)
-            };
-            self.chain.block_times_cache.write().set_time_observed(
-                hash,
-                slot,
-                seen_timestamp,
-                None,
-                None,
-            );
+                // Trigger processing for work referencing this block.
+                let reprocess_msg = ReprocessQueueMessage::BlockImported {
+                    block_root: *hash,
+                    parent_root,
+                };
+                if reprocess_tx.try_send(reprocess_msg).is_err() {
+                    error!(self.log, "Failed to inform block import"; "source" => "rpc", "block_root" => %hash)
+                };
+                self.chain.block_times_cache.write().set_time_observed(
+                    *hash,
+                    slot,
+                    seen_timestamp,
+                    None,
+                    None,
+                );
 
-            self.chain.recompute_head_at_current_slot().await;
+                self.chain.recompute_head_at_current_slot().await;
+            }
+            Ok(AvailabilityProcessingStatus::MissingComponents(..)) => {
+                self.fetch_engine_blobs_and_publish(signed_beacon_block, block_root)
```

Comment:
Please add a detailed comment on why `fetch_engine_blobs_and_publish` is triggered here and only here

Reply:
It's also triggered gossip as well, I'll add a comment
https://github.com/sigp/lighthouse/blob/662d6cf931eb2dca8df90f47924a13c2cc4a60c0/beacon_node/network/src/network_beacon_processor/gossip_methods.rs#L1519

Reply:
Done